### PR TITLE
chore: fix clippy warnings and add CI

### PR DESCRIPTION
**Pull request by vmx**

### CODE REVIEW

File: tests/ipld.rs [hunk 1]
```diff
@@ -24,11 +24,12 @@ mod std_tests {
 
     use std::iter::FromIterator;
 
+    #[allow(clippy::useless_format)]
     #[test]
     fn serde() {
         let tuple_struct = TupleStruct(format!("test"), -60, 3000);
 
-        let tuple = (format!("hello"), -50.0040957, -12.094635556478);
```
Comment (MarcoPolo) [inline tests/ipld.rs:31]: I'm just curious, was this a lint (the number change)?

**Review by MarcoPolo — APPROVED** [review]

File: tests/ipld.rs (hunk 1 above)
Comment (Stebalien) [reply tests/ipld.rs:31]: Probably because rust can't accurately represent it?

Comment (Stebalien) [reply tests/ipld.rs:31]: (that's a total guess)

**Review by Stebalien — APPROVED** [review]

File: tests/ipld.rs (hunk 1 above)
Comment (vmx) [reply tests/ipld.rs:31]: Yes, the lint was: https://rust-lang.github.io/rust-clippy/master/index.html#excessive_precision

**Pull request merged**